# Thaitanic

Goal is to generate a dataset and A/B test it.

I estimated 2019 data because it felt a little ridiculous to estimate 2020 data given covid. 

This is based on an actual restaurant that I wrote a case study about for a business fraternity in 2018. (Everything in the case is real except for the backstory). 

### Notes

Thaitanic menu: http://thaitanicstreetfood.com/wp-content/pdf/ThaiTanic%20Streetfood%20Menu.pdf

Thaitanic site: http://thaitanicstreetfood.com/

Thaitanic yelp: https://www.yelp.com/biz/thaitanic-streetfood-sausalito-2

Yelp robots.txt: https://www.yelp.com/robots.txt

Case study doc: https://docs.google.com/document/d/1iwT_TimRU4odQRN2J2ChMyN4kJbqcM5lxDdTyFkDV0U/edit

Case study notes: https://docs.google.com/document/d/1uOokFU3q-R25ZbLXpE8LHrUBQi1ky6k2tBpA3ejNHvI/edit



# Import libraries

In [120]:
import pandas as pd
import numpy as np
import datetime
import random
import calendar

In [154]:
# maybe this should just be split into telling if it's lunch or dinner...

def get_weekday_weight(weekday_value):
    # takes a weekday and selects lunch or dinner based on popular_times, then returns a weight for the day. returns zero on mondays (store is closed)
    if weekday_value == 1:
        weekday = "tuesday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    elif weekday_value == 2:
        weekday = "wednesday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    elif weekday_value == 3:
        weekday = "thursday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    elif weekday_value == 4:
        weekday = "friday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    elif weekday_value == 5:
        weekday = "saturday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    elif weekday_value == 6:
        weekday = "sunday"
        day_type = 'lunch'
        if random.random() <= service_hours[f'{weekday}_lunch'][0]/(service_hours[f'{weekday}_lunch'][0]+service_hours[f'{weekday}_dinner'][0]):# this percentage is based on their ratio...
            day_weight = service_hours[f'{weekday}_lunch'][3]
            return day_type, day_weight
        else:
            day_type = 'dinner'
            day_weight = service_hours[f'{weekday}_dinner'][3]
            return day_type, day_weight
    else:
        weekday = "monday"
        day_type = "none"
        day_weight = 0
        return day_type, day_weight
        
test_type, test_weight = get_weekday_weight(2)
print(test_type)
print(test_weight)

dinner
0.71875


In [142]:
summer_months = [6, 7, 8]
winter_months = [11, 12]

### yeah, i think the move is to just make it a multiplier on a standard orders distribution...

def get_month_weight(month_value): # think about if you want to use a percentage or a value here. percent probably cleaner.
    # takes a month value and returns a weighted value
    if month_value in summer_months:
#         orders = int(np.random.normal(loc=270*days_per_month.get(month), scale=600))
        month_weight = 0.8
        return month_weight
    elif month_value in winter_months:
#         orders = int(np.random.normal(loc=300*days_per_month.get(month), scale=600))
        month_weight = 0.9
        return month_weight
    else:
#         orders = int(np.random.normal(loc=250*days_per_month.get(month), scale=600))
        month_weight = 0.75
        return month_weight
    
print(get_month_weight(1))

0.75


In [123]:
# probably can just turn that into its own formula

def calculate_daily_weights(weight):
    return (50 + 25 * weight / 4) / 100

service_hours = { # these are in google values - how do they correlate to our weights?
    # weekday_time: [weight, weight/sum_of_weights, opening, closing]
    "tuesday_lunch": [3, 11, 14],
    "tuesday_dinner": [3.5, 17, 20],
    "wednesday_lunch": [1.5, 11, 14],
    "wednesday_dinner": [3.5, 17, 20],
    "thursday_lunch": [2.5, 11, 14],
    "thursday_dinner": [4, 17, 20],
    "friday_lunch": [3, 11, 14],
    "friday_dinner": [7, 17, 20],
    "saturday_lunch": [7.5, 12, 15],
    "saturday_dinner": [9, 17, 20],
    "sunday_lunch": [6.5, 12, 15],
    "sunday_dinner": [6, 17, 20],
}

for service in service_hours.values():
    if len(service) == 3:
        service.append(calculate_daily_weights(service[0]))

print(service_hours)

{'tuesday_lunch': [3, 11, 14, 0.6875], 'tuesday_dinner': [3.5, 17, 20, 0.71875], 'wednesday_lunch': [1.5, 11, 14, 0.59375], 'wednesday_dinner': [3.5, 17, 20, 0.71875], 'thursday_lunch': [2.5, 11, 14, 0.65625], 'thursday_dinner': [4, 17, 20, 0.75], 'friday_lunch': [3, 11, 14, 0.6875], 'friday_dinner': [7, 17, 20, 0.9375], 'saturday_lunch': [7.5, 12, 15, 0.96875], 'saturday_dinner': [9, 17, 20, 1.0625], 'sunday_lunch': [6.5, 12, 15, 0.90625], 'sunday_dinner': [6, 17, 20, 0.875]}


### show me the maximum and minimums!

look for loc time smallest day weight times smallest month weight.

then we'll factor in variance for 20 times smallest day weight times smallest month weight.

it looks like the largest possible gets to our estimated max capacity, and the smallest possible is about half of that. seems fine to me.

In [11]:
service_hours

{'tuesday_lunch': [3, 11, 14, 0.6875],
 'tuesday_dinner': [3.5, 17, 20, 0.71875],
 'wednesday_lunch': [1.5, 11, 14, 0.59375],
 'wednesday_dinner': [3.5, 17, 20, 0.71875],
 'thursday_lunch': [2.5, 11, 14, 0.65625],
 'thursday_dinner': [4, 17, 20, 0.75],
 'friday_lunch': [3, 11, 14, 0.6875],
 'friday_dinner': [7, 17, 20, 0.9375],
 'saturday_lunch': [7.5, 12, 15, 0.96875],
 'saturday_dinner': [9, 17, 20, 1.0625],
 'sunday_lunch': [6.5, 12, 15, 0.90625],
 'sunday_dinner': [6, 17, 20, 0.875]}

In [28]:
# lowest possible value per day
print(f" lowest possible value per day: {(330 * 0.65625 * 0.75) - (20 * 0.65625 * 0.75)}")
print(f" lowest possible value per month: {((330 * 0.65625 * 0.75) - (20 * 0.65625 * 0.75))*28}")


 lowest possible value per day: 152.578125
 lowest possible value per month: 4272.1875


In [31]:
# highest possible value per day (actaul average is much lower because only two months can possible reach this)

print(f" highest possible value per day: {(330 * 1.0625 * 0.90) + (20 * 1.0625 * 0.9)}")
print(f" highest possible value per month: {((330 * 1.0625 * 0.90) + (20 * 1.0625 * 0.9))*31}")

 highest possible value per day: 334.6875
 highest possible value per month: 10375.3125


In [ ]:
# maybe later also figure out the expected range for the dataset to produce

# Create data

create a dataframe of products with frequencies based on weights.

we'll make some months have more than others (google says 72% of restaurants have their highest in november/december (2018): https://www.thinkwithgoogle.com/consumer-insights/consumer-trends/restaurant-sale-statistics-by-month).

seems like it varies heavily depending on the type of restaurant and the location, but probably there's also a general increase in the summer.

In [18]:
products = {
    # Product: [Price, Weight]
    'Dish 1': [12, 10],
    'Dish 2': [13, 4],
    'Dish 3': [13, 8],
    'Dish 4': [15, 6],
    'Dish 5': [12, 9],
    'Dish 6': [13, 2],
    'Dish 7': [11, 7],
}

columns = ['Order ID', 'Product', 'Quantity Ordered', 'Price Each', 'Order Date']

In [186]:
%%time

# trying to create from very simple

years = [2019] 
months = {
    1: "January",
    2: "February",
}

# factor in relative weight
product_list = [product for product in products]
weights = [products[product][1] for product in products]

order_id = 3910

for year in years:
    
    for month in months:
        month_df = pd.DataFrame(columns=columns)
        month_weight = get_month_weight(month)
        
        for day in range(1,calendar.monthrange(year, month)[1]+1):
            date = datetime.datetime(year, month, day)
            day_weight = get_weekday_weight(date.weekday())[1]
            orders = int(np.random.normal(loc=330*day_weight*month_weight, scale = 20*day_weight*month_weight))
            
            for i in range(orders):
                product = random.choices(product_list, weights=weights)[0]
                price = products[product][0]
                
                day_type = get_weekday_weight(date.weekday())[0]
                # all of the times are still being put in the same lunch or dinner for each day...
                random_hour = random.randint(get_open_hours(date.weekday(), day_type)[0], get_open_hours(date.weekday(), day_type)[1]) # day_type is generated by day, not by order
                random_minute = random.randint(0, 59)
                order_date = datetime.datetime(year, month, day, random_hour, random_minute)
                order_day_type, order_day_weight = get_weekday_weight(order_date.weekday())
                
                day_df = pd.DataFrame(columns=columns)
                day_df.loc[i] = [order_id, product, 1, price, order_date]
                
                month_df = month_df.append(day_df)
                order_id += 1
                
        month_df.to_csv(f'{months[month]}_{year}_test_data.csv')
        print(f"{months[month]} {year} complete!")

January 2019 complete!
February 2019 complete!
Wall time: 1min 7s


In [175]:
def get_open_hours(day_of_week, day_type): # remind me to standardize variable nomenclature
    # takes a weekday and returns lunch hours. No need to do this for dinner hours because they're always the same.
    weekdays = {
        1: "tuesday",
        2: "wednesday",
        3: "thursday",
        4: "friday"
    }
    weekends = {
        5: "saturday",
        6: "sunday"
    }
    if day_of_week in weekdays.keys():
        if day_type == "lunch":
            opening, closing = 11, 14
            return opening, closing
        else:
            opening, closing = 17, 20
            return opening, closing
    elif day_of_week in weekends: # probably safe to just use else here
        if day_type == "lunch":
            opening, closing = 12, 15
            return opening, closing
        else:
            opening, closing = 17, 20
            return opening, closing
    else: # mondays - i don't think it ever reaches this point
        print('WARNING: MONDAY DETECTED')
        opening, closing = 0, 0
        return opening, closing
        
print(get_open_hours(5, 'lunch'))

(12, 15)


In [160]:
weekdays = {
    1: "tuesday",
    2: "wednesday",
    3: "thursday",
    4: "friday"
}
weekends = {
    5: "saturday",
    6: "sunday"
}
print(weekdays.keys())

dict_keys([1, 2, 3, 4])


In [183]:
df = pd.read_csv('January_2019_test_data.csv', index_col = False)
df[800:850]

Unnamed: 0  Order ID Product  Quantity Ordered  Price Each  \
800          71      4710  Dish 7                 1          11   
801          72      4711  Dish 4                 1          15   
802          73      4712  Dish 4                 1          15   
803          74      4713  Dish 5                 1          12   
804          75      4714  Dish 7                 1          11   
805          76      4715  Dish 5                 1          12   
806          77      4716  Dish 1                 1          12   
807          78      4717  Dish 6                 1          13   
808          79      4718  Dish 7                 1          11   
809          80      4719  Dish 7                 1          11   
810          81      4720  Dish 4                 1          15   
811          82      4721  Dish 6                 1          13   
812          83      4722  Dish 4                 1          15   
813          84      4723  Dish 1                 1          12   
814          85      4724  Dish 4                 1          15   
815          86      4725  Dish 1                 1          12   
816          87      4726  Dish 5                 1          12   
817          88      4727  Dish 4                 1          15   
818          89      4728  Dish 1                 1          12   
819          90      4729  Dish 5                 1          12   
820          91      4730  Dish 2                 1          13   
821          92      4731  Dish 2                 1          13   
822          93      4732  Dish 7                 1          11   
823          94      4733  Dish 5                 1          12   
824          95      4734  Dish 5                 1          12   
825          96      4735  Dish 2                 1          13   
826          97      4736  Dish 3                 1          13   
827          98      4737  Dish 1                 1          12   
828          99      4738  Dish 1                 1          12   
829         100      4739  Dish 2                 1          13   
830         101      4740  Dish 3                 1          13   
831         102      4741  Dish 5                 1          12   
832         103      4742  Dish 3                 1          13   
833         104      4743  Dish 4                 1          15   
834         105      4744  Dish 6                 1          13   
835         106      4745  Dish 2                 1          13   
836         107      4746  Dish 3                 1          13   
837         108      4747  Dish 2                 1          13   
838         109      4748  Dish 3                 1          13   
839         110      4749  Dish 5                 1          12   
840         111      4750  Dish 5                 1          12   
841         112      4751  Dish 5                 1          12   
842         113      4752  Dish 1                 1          12   
843         114      4753  Dish 3                 1          13   
844         115      4754  Dish 7                 1          11   
845         116      4755  Dish 7                 1          11   
846         117      4756  Dish 3                 1          13   
847         118      4757  Dish 3                 1          13   
848         119      4758  Dish 3                 1          13   
849         120      4759  Dish 5                 1          12   

              Order Date  
800  2019-01-05 13:13:00  
801  2019-01-05 13:54:00  
802  2019-01-05 15:29:00  
803  2019-01-05 12:36:00  
804  2019-01-05 14:12:00  
805  2019-01-05 15:00:00  
806  2019-01-05 14:51:00  
807  2019-01-05 12:57:00  
808  2019-01-05 13:50:00  
809  2019-01-05 15:52:00  
810  2019-01-05 13:49:00  
811  2019-01-05 15:33:00  
812  2019-01-05 15:55:00  
813  2019-01-05 12:29:00  
814  2019-01-05 14:04:00  
815  2019-01-05 14:18:00  
816  2019-01-05 13:18:00  
817  2019-01-05 14:10:00  
818  2019-01-05 13:38:00  
819  2019-01-05 14:26:00  
820  2019-01-05 15:5

# below not useful anymore

In [3]:
# select select a day

def generate_random_time(month): # probably add back opening and closing times later
    # generate a random day in the given month, within restaurant hours
    year = 2019
    random_day = random.randint(1, calendar.monthrange(2019, month)[1])
    
    opening_time = 11
    closing_time = 14
    
    random_hour = random.randint(opening_time, closing_time) # should i add a light normal distribution here?
    random_minute = random.randint(1, 59)
    
    date = datetime.datetime(year, month, random_day, random_hour, random_minute)
    return date
    
print(generate_random_time(1))

2019-01-02 11:58:00


In [4]:
summer_months = [6, 7, 8]
winter_months = [11, 12]
days_per_month = {
    1: 31, 2: 28, 3: 31,
    4: 30, 5: 31, 6: 30,
    7: 31, 8: 31, 9: 30,
    10: 31, 11: 30, 12: 31
}

product_list = [product for product in products]
weights = [products[product][1] for product in products]

order_id = 3910

for month in days_per_month.keys(): # probably a better way to select months, right?
    if month in summer_months:
        orders = int(np.random.normal(loc=270*days_per_month.get(month), scale=600))
    elif month in winter_months:
        orders = int(np.random.normal(loc=300*days_per_month.get(month), scale=600))
    else:
        orders = int(np.random.normal(loc=250*days_per_month.get(month), scale=600))
        
    df = pd.DataFrame(columns=columns)

    for i in range(orders):
        date = generate_random_time(month)
        
        product = random.choices(product_list, weights=weights)[0]
        price = products[product][0]
        df.loc[i] = [order_id, product, 1, price, date]
        order_id +=1 
    
    month_name = calendar.month_name[month]
    df.to_csv(f'{month_name}_data.csv')
    print(f'{month_name} has been generated!')
    break

January has been generated!


### have to explain how i got the weights...

i took the bottom line of business to be 50%, the first line to be 75%, the third line to be 100%, and the last line to be 100% with a wait.

the restaurant is open for four hours at a time, so i took the average bar per lunch or dinner. for example, during tuesday lunch the average bar is about 3/4 of the way between the bottom line (50%) and the first line (75%). so 50% + 25\*0.75 = 68.75%, which is the weight i used for tuesday lunch.

##### it's just 50% + the average as a percentile

### how much should the day affect the overall sales?

i have weights, but they're in google popular times ratio form, which isn't relevant to my weights yet, i think.

i can probably take the total number of rows generated... and then divide up the days based on a ratio? does there need to be any randomness there?

i guess i would use an if statement... with random weights based on the weekday/hour for each?

we can check what day of the week it is using date.weekday()

the downside is that if there's a month with five of one day instead of four, sales will be split across five days that month instead of four..

# is it worth it to generate one day at a time and then combine them into a month and save that?

In [50]:
'''
summer_months = [6, 7, 8]
winter_months = [11, 12]
days_per_month = {
    1: 31, 2: 28, 3: 31,
    4: 30, 5: 31, 6: 30,
    7: 31, 8: 31, 9: 30,
    10: 31, 11: 30, 12: 31
}

product_list = [product for product in products]
weights = [products[product][1] for product in products]

order_id = 3910

for month in days_per_month.keys(): # probably a better way to select months, right?
    if month in summer_months:
        orders = int(np.random.normal(loc=270*days_per_month.get(month), scale=600))
    elif month in winter_months:
        orders = int(np.random.normal(loc=300*days_per_month.get(month), scale=600))
    else:
        orders = int(np.random.normal(loc=250*days_per_month.get(month), scale=600))
        
    df = pd.DataFrame(columns=columns)

    for i in range(orders):
        date = generate_random_time(month)
        
        product = random.choices(product_list, weights=weights)[0]
        price = products[product][0]
        df.loc[i] = [order_id, product, 1, price, date]
        order_id +=1 
    
    month_name = calendar.month_name[month]
    df.to_csv(f'{month_name}_data.csv')
    print(f'{month_name} has been generated!')
    break
'''
years = [2019] # maybe add some tiny multiplier to each, probably increasing per year...
months = {
    1: "January",
#     2: "February",
}

order_id = 3910

for year in years:
    
    for month in months:
        month_df = pd.DataFrame(columns=columns)
        
        for day in range(1,calendar.monthrange(year, month)[1]):
            random_hour = random.randint(11, 14) # opening_time, closing_time
            random_minute = random.randint(0, 59)
            date = datetime.datetime(year, month, day, random_hour, random_minute)
            
            if date.weekday() == 1:
                if random.random() <= service_hours['tuesday_lunch'][0]/(service_hours['tuesday_lunch'][0]+service_hours['tuesday_dinner'][0]):# this percentage is based on their ratio...
                    # it's lunch
                    # create a variable with the tuesday lunch multiplier
                    day_weight = service_hours['tuesday_lunch'][3]
                else:
                    # it's dinner
                    # create a variable with the tuesday dinner multiplier
                    day_weight = service_hours['tuesday_dinner'][3]
            else:
                print("Skip (unfinished day)")
#             elif date.weekday() = 2: # continue for all six days...
#                 if random.random() <= 0.5: 
#                     # it's lunch
#                     # create a variable with the tuesday lunch multiplier
#                 else:
#                     # it's dinner
#                     # create a variable with the tuesday dinner multiplier
            
            if month in summer_months: # where do i generate the data?
                orders = int(np.random.normal(loc=270*day_weight, scale=20)) # may want to reduce monthly variation and check scale
            elif month in winter_months:
                orders = int(np.random.normal(loc=300*day_weight, scale=20))
            else:
                orders = int(np.random.normal(loc=250*day_weight, scale=20))
            
            for i in range(orders):
                date = generate_random_time(month)
        
                product = random.choices(product_list, weights=weights)[0]
                price = products[product][0]
                df.loc[i] = [order_id, product, 1, price, date]
                order_id +=1 
            
            
            
            
            
            print(date)
            print(date.weekday())
            print()
                
            # now i know what day it is. 
            # need to figure out if it's lunch or dinner
            
            # let's say it's tuesday: 3/6.5 it's lunch, 3.5/6.5 it's dinner.
            # of the orders generated, what percentage goes to lunch, and what percentage goes to dinner?
            # if random_number <= 3, it's morning. else dinner.
                
            # append to month df
            # save month df

    
    
        

2019-01-28 14:04:00
0

Skip (unfinished day)
2019-01-16 14:35:00
2

Skip (unfinished day)
2019-01-02 13:01:00
2

Skip (unfinished day)
2019-01-20 13:53:00
6

Skip (unfinished day)
2019-01-27 11:26:00
6

Skip (unfinished day)
2019-01-14 11:15:00
0

Skip (unfinished day)


KeyboardInterrupt: 

# try even more simple

just make it work from a days perspective

In [13]:
service_hour

NameError: name 'service_hour' is not defined

In [ ]:
calendar.monthrange(2020, 2)

### what should the day multiplier be? (how much should the day affect sales?)

be careful because we already have a month multiplier in effect... (do a sanity check calculation on this later)

what affects a day more, the month that it's in, or the day of the week that it is? probably the day of the week that it is, by a bit.

if 330 is the highest realistic demand per day, all mulitpliers should be lower (with some randomness thrown in for good measure). 

i think i just want to look at the popular times thing. it's a bit hard because google just gives us some bars with no numbers, but let's say two bars means the place is at max capacity, whereas one bar means they're at 75% capacity. maybe zero bars means 50%. 

i compared the popular times bars to some other places i've eaten like in n out or some brunch places for good measure. i think covid is actually these numbers because fast casual places with drive throughts seem to have high numbers, whereas sit down places are still lower. i'll be a bit generous with the data and just call the bars how i mentioned. it's more of an eyeballing thing though.

i'm not sure how accurate the popular times is though, since a lot of the busier places (even with almost max bars) had an additional indicator that there was no wait...

for reference, in n out is estimated to do around 500 orders a day, so a sit down restaurant serving closer to 300 including takeout seems reasonable to me.

i'll just use the average per service hour which would spread the times out evenly, even though they peak closer to the middle (maybe can fix that later?).

### Estimate dine-in/takeout sales 

Eyeballing the pictures, it looks like there are 4 + 4 + 2 + 2 + 2 + 3 indoor seats and 4 + 4 outdoor seats.

##### can i trim these links?

Google Earth to thai tanic (street view): https://earth.google.com/web/search/ThaiTanic+Streetfood,+Bridgeway,+Sausalito,+CA/@37.85851259,-122.48337616,3.31408215a,0d,60y,258.6686533h,78.86489434t,0r/data=CpkBGm8SaQolMHg4MDg1ODQ1MWI0ZjE4YTkxOjB4YTU2MWViNDU4NTcyMWU5ZBl0mfW94e1CQCG4HK9A9J5ewCouVGhhaVRhbmljIFN0cmVldGZvb2QsIEJyaWRnZXdheSwgU2F1c2FsaXRvLCBDQRgBIAEiJgokCUlFdQiQmTZAEUdFdQiQmTbAGda0jUD-tz1AIQtYmNuq1VLAIhoKFlNMblNLNWVpQURnbUg4SlN3MzZoTmcQAg

Google Earth to thai tanic (aerial view)
https://earth.google.com/web/search/ThaiTanic+Streetfood,+Bridgeway,+Sausalito,+CA/@37.85845058,-122.4836834,2.8947445a,41.80733363d,35y,-45.474136h,27.45597896t,360r/data=CpkBGm8SaQolMHg4MDg1ODQ1MWI0ZjE4YTkxOjB4YTU2MWViNDU4NTcyMWU5ZBl0mfW94e1CQCG4HK9A9J5ewCouVGhhaVRhbmljIFN0cmVldGZvb2QsIEJyaWRnZXdheSwgU2F1c2FsaXRvLCBDQRgBIAEiJgokCUlFdQiQmTZAEUdFdQiQmTbAGda0jUD-tz1AIQtYmNuq1VLA

I also see some "popular times" values provided by Google - these are for 2020 but seem pretty reasonable for a restaurant. I'm just ballparking so this looks good - the popular times even reflect the different opening hours between weekdays and weekends. This is 2020 data but it looks general enough that I feel comfortably applying it to 2019.

##### options for estimating demand

13ish indoor seats, and 8 outdoor seats.

Let's assume indoors vary around 90% occupancy all the time, and maybe the outdoor seats are closer to 60% occupancy. These are probably a bit generous. 

(?) How can I use the weights from the Google data? I can use the lowest bar to estimate 90% occupancy and throw in some randomness. So if the popularity looks close to the first bar, I'll say 90% occupancy. 

##### we can ballpark their revenue range 

by estimating average meal price * number of meals served. let's just assume they do half as much delivery/takeout as they serve regularly?

it's a \$\$ restaurant, which appears to be the \\$11-\\$20 range (source: quora https://www.quora.com/What-exactly-do-the-price-indicators-mean-in-Google-restaurant-listings-and-does-it-depend-on-the-part-of-the-country)

I think that feels a bit low looking at their menu though, so I'm okay estimating $20 average per person.

they're open 8 hours a day, 6 days a week. they have 13 indoor seats and 8 outdoor seats, i don't think outdoor is as popular so i'm okay just putting that as 20 seats. + 10 seats equivalent for delivery is 30.

average meal is like 45 minutes? 30 feels short, although a lot of the meals are during lunch hour. 8/0.75 is 10.66, let's just round that to 11.

so 30 meals per hour * 11 meals per day * \\$20 per meal * 6 days per week  * 50 weeks = \\$1,980,000/year in revenue. these numbers sound reasonable, so i'm okay with using 


In [ ]:
30 * 11 * 30

so volume is like 330 / day? is that low?

we'll just call it 10k/month. the 10k represents out-the-door demand. like if there was a two hour wait at all times for this place.

In [ ]:
330 * 31

In [ ]:
330 * 28

In [ ]:
days_per_month = {
    1: 31, 2: 28, 3: 31,
    4: 30, 5: 31, 6: 30,
    7: 31, 8: 31, 9: 30,
    10: 31, 11: 30, 12: 31
}

### break down logic from the top

(!) daily variance (break down by daily average, not sure if we need hourly). 


(!) monthly variance (max occupancy all day is 330/330 - this is what you can realistically serve if you have infinite demand)
- in november/december, we'll do 30 mean (90%) occupancy
- in june/july/august, we'll do 270/330 (about 80%) occupancy
- and every other month we'll do 250/330 (75%) occupancy.

In [ ]:
df = pd.read_csv('January_data.csv')